en/dev_setup/dev_env_windows_wsl.md

```diff
@@ -0,0 +1,270 @@
+   - Type `cmd` and press **Enter** to open the prompt.
+1. To start WSL and access the WSL shell, execute the command:
+   ```
+   wsl
+   ```
+
+:::note
+Enter the following commands to first close the WSL shell, and then shut down WSL:
+```
+exit
+wsl --shutdown
+````
+Alternatively, after entering `exit` you can just close the prompt.
+::: 
+
+### Install PX4 Toolchain
+
+Next we download the PX4 source code within the WSL2 environment, and use the normal *Ubuntu installer script* to to set up the developer environment.
+This will install the toolchain for Gazebo simulation, JMAVSim simulation and Pixhawk/NuttX hardware.
+
+To install the development toolchain:
+
+1. [Open a WSL2 Shell](#opening-a-wsl-shell) (if it is still open you can use the same one that was used to install WSL2).
+1. Execute the command `cd ~` to switch to the home folder of WSL for the next steps.
+   :::warning
+   This is important!
+   If you work from a location outside of the WSL file system you'll run into issues such as very slow execution and access right/permission errors.
+   :::
+
+1. Download PX4 source code using `git` (which is already installed in WSL2):
+   ```bash
+   git clone https://github.com/PX4/PX4-Autopilot.git --recursive
+   ```
+1. Run the **ubuntu.sh** installer script and acknowledge any prompts as the script progresses:
+   ```bash
+   bash ./PX4-Autopilot/Tools/setup/ubuntu.sh
+   ```
+
+   :::note
+   This installs tools to build PX4 for Pixhawk, Gazebo and JMAVSim targets.
+   - You can use the `--no-nuttx` and `--no-sim-tools` options to omit the NuttX and/or simulation tools.
+   - Other Linux build targets are untested (you can try these by entering the appropriate commands in [Ubuntu Development Environment](../dev_setup/dev_env_linux_ubuntu.md) into the WSL shell).
+   :::
+
+1. Restart the "WSL computer" after the script completes (exit the shell, shutdown WSL, and restart WSL):
+   ```
+   exit
+   wsl --shutdown
+   wsl
+   ```
+1. Switch to the PX4 repository in the WSL home folder:
+   ```
+   cd ~/PX4-Autopilot
+   ```
+1. Build the PX4 SITL target and test your environment:
+   ```
+   make px4_sitl
+   ```
+   
+For more build options see [Building PX4 Software](../dev_setup/building_px4.md).
+
+
+## Visual Studio Code Integration
+
+VS Code running on Windows is well integrated with WSL.
+
+To set up the integration:
+1. [Download](https://code.visualstudio.com/) and install Visual Studio Code (VS Code) on Windows,
+2. Open _VS Code_.
+3. Install the extension called [Remote - WSL](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-wsl) (marketplace) 
+4. [Open a WSL shell](#opening-a-wsl-shell)
+5. In the WSL shell, switch to the PX4 folder:
+   ```
+   cd ~/PX4-Autopilot
+   ```
+6. In the WSL shell, start VS Code:
+   ```
+   code .
+   ```
+   This will open the IDE fully integrated with the WSL shell.
+   
+   Make sure you always open the PX4 repository in the Remote WSL mode.
```

Comment:
FYI, @MaEtUgR Thanks for your explanation of the WSL integration.

To make this clear to others, I have added this line here - so that make it very clear that "this" is what it mean to open the PX4 repository in the Remote WSL mode.

Then in step 7 I add your image of how you "Open Recent" and not that it is also " opening in Remote WSL mode "

Reply:
Perfect, thanks a lot. I just didn't see how to tell that in a clear way 😇